File: source java/organism/plants/Belladonna.java
package organism.plants;

import java.awt.Color;
import organism.CollisionResult;
import organism.Organism;
import organism.manager.Organisms;
import utilities.position.Position;
import utilities.savingsystem.Loader;
import virtualworld.World;

public class Belladonna extends Plant
{
    private static final int STRENGTH = 99;

    
    public Belladonna(Position position, World world) 
    {
        super(STRENGTH, position, world);
    }
    
    public Belladonna(Loader loader, World world, Position position)
    {
        super(loader, world, position);
    }
    
    @Override
    public CollisionResult Collide(Organism organism)
    {
        organism.Die();
        GetWorld().AddToLogs(organism.toString() + " dies after eating " + toString());
        return CollisionResult.PLANT;
    }
    
    @Override
    public Organisms GetEnum() 
    {
        return Organisms.BELLADONNA;
    }
    
    @Override
    public Color GetColor()
    {
        return Color.red;
    }
    
    @Override
    public String toString()
    {
        return "Belladonna";
    }
}

File: source python/Organisms/Plants/Belladonna.py
from Application.Organisms.Plants.Plant import Plant
from Application.Organisms.Organism import ColisionResult
from Application.Utilities import Colors


class Belladonna(Plant):

    STRENGTH = 99

    def __init__(self, position, world, **kwargs):
        super().__init__(Belladonna.STRENGTH, position, world, **kwargs)

    def Collide(self, organism):
        organism.Die()
        print(str(organism) + " dies after eating " + str(self))
        return ColisionResult.PLANT

    @staticmethod
    def __str__():
        return "Belladonna"

    def GetColor(self):
        return Colors.RED

    def GetLetter(self):
        return "J"